### ISSUE TITLE
Using ES2018 object rest syntax to default a function parameter when targeting ES5 outputs incorrectly

### ISSUE THREAD

**Issue by eliw00d** [microsoft/TypeScript#47431]
# Bug Report

When setting target to ES5 using object rest syntax to default a function parameter I get output that errors trying to access the rest variable.

### 🔎 Search Terms

object rest default parameter ES5 output

### 🕗 Version & Regression Information

- I was unable to test this on prior versions because I only ran across it while adding code with 4.5.4 installed.

### ⏯ Playground Links

[ES5 target](https://www.typescriptlang.org/play?target=1&ts=4.5.5#code/MYewdgzgLgBAZgVzMGBeGAKA3jA+gSwBMAaGAOgoQgFMAnGAXwC4YcDCXpb8wBzUsAEMAttU5RufRqVEQIg3tTQwABgBIsVOmSGiGMBAAdCgqNUIwICYMGpzEAGwcBPFQEo0APlYBYAFAwMKCQIA7UZA4gvBiy8opu-gz+-ojI2OwsAOQAjJkCImIwmQAKgoIAbiCZDG5AA) incorrectly throws `Cannot read properties of undefined (reading 'name')` error

[ES2018 target](https://www.typescriptlang.org/play?target=5&ts=4.5.5#code/MYewdgzgLgBAZgVzMGBeGAKA3jA+gSwBMAaGAOgoQgFMAnGAXwC4YcDCXpb8wBzUsAEMAttU5RufRqVEQIg3tTQwABgBIsVOmSGiGMBAAdCgqNUIwICYMGpzEAGwcBPFQEo0APlYBYAFAwMKCQIA7UZA4gvBiy8opu-gz+-ojI2OwsAOQAjJkCImIwmQAKgoIAbiCZDG5AA) correctly logs `"Paavo updated successfully"` to console

### 💻 Code

Playground link should suffice

### 🙁 Actual behavior

ES5 output of ES2018 code throws error.

### 🙂 Expected behavior

ES5 output of ES2018 code should not throw error.

**Comment by eliw00d** [comment]
```javascript
var func = function (_a, message) {
    if (message === void 0) { message = "".concat(user.name, " updated successfully"); }
    var _id = _a._id, user = __rest(_a, ["_id"]);
    console.log(message);
};
```
should probably be:
```javascript
var func = function (_a, message) {
    var _id = _a._id, user = __rest(_a, ["_id"]);
    if (message === void 0) { message = "".concat(user.name, " updated successfully"); }
    console.log(message);
};
```
or:
```javascript
var func = function (_a, message) {
    var user = __rest(_a, ["_id"]);
    if (message === void 0) { message = "".concat(user.name, " updated successfully"); }
    var _id = _a._id;
    console.log(message);
};
```

**Comment by eliw00d** [comment]
[https://www.typescriptlang.org/play?target=1&ts=4.6.2#code/MYewdgzgLgBAZgVzMGBeGAKA3jA+gSwBMAaGAOgoQgFMAnGAXwC4YcDCXpb8wBzUsAEMAttU5RufRqVEQIg3tTQwABgBIsVOmSGiGMBAAdCgqNUIwICYMGpzEAGwcBPFQEo0APlYBYAFAwMKCQIA7UZA4gvBiy8opu-gz+-ojI2OwsAOQAjJkCImIwmQAKgoIAbiCZDG5AA](ES5 target) correctly logs "Paavo updated successfully" to console as of 4.6.x

**Comment by RyanCavanaugh** [comment]
@eliw00d to confirm: This is fixed and we should close it?

**Comment by eliw00d** [comment]
Correct!

**Comment by RyanCavanaugh** [comment]
Thanks 🙂
